contactsummary/js/src/contact-summary/controllers/keyDetails.js

```diff
@@ -14,37 +15,52 @@ define([
    * @param {ContractService} Contract
    * @constructor
    */
-  function KeyDetailsCtrl($log, ContactDetails, Contract) {
+  function KeyDetailsCtrl($log, ContactDetails, Contract, pubSubService) {
     $log.debug('Controller: KeyDetailsCtrl');
 
     this.ready = false;
 
-    ContactDetails.get()
-      .then(function (response) {
-        this.contactDetails = response;
-        return Contract.getPrimary();
-      }.bind(this))
-      .then(function (response) {
-        if (_.isEmpty(response)) {
-          return;
+
+
+    var getContacts = function(){
```

Comment:
I used first one so that bind can be used. Otherwise another line would be needed to bind
``` 
function getContacts() {}
getContacts.bind(this);
```